analytics_data_api/management/commands/generate_data.py

```diff
@@ -0,0 +1,339 @@
+
+            for country_code, ratio in country_ratios.items():
+                count = int(ratio * daily_total)
+                models.CourseEnrollmentByCountry.objects.using(database).create(course_id=course_id, date=date,
+                                                                                count=count,
+                                                                                country_code=country_code)
+            if add_birth_year:
+                for birth_year, ratio in birth_years.items():
+                    count = int(ratio * daily_total)
+                    models.CourseEnrollmentByBirthYear.objects.using(database).create(course_id=course_id, date=date,
+                                                                                      count=count,
+                                                                                      birth_year=birth_year)
+
+            progress.update(1)
+            date = date + datetime.timedelta(days=1)
+
+        models.CourseMetaSummaryEnrollment.objects.using(database).filter(course_id=course_id).delete()
+        for index, (mode, ratio) in enumerate(enrollment_mode_ratios.items()):
+            count = int(ratio * daily_total)
+            pass_rate = min(random.normalvariate(.45 + (.1 * index), .15), 1.0)
+            cumulative_count = count + random.randint(0, 100)
+            models.CourseMetaSummaryEnrollment.objects.using(database).create(
+                course_id=course_id, catalog_course_title='Demo Course', catalog_course='Demo_Course',
+                start_time=timezone.now() - datetime.timedelta(weeks=6),
+                end_time=timezone.now() + datetime.timedelta(weeks=10),
+                pacing_type='self_paced', availability='Starting Soon', enrollment_mode=mode, count=count,
+                cumulative_count=cumulative_count, count_change_7_days=random.randint(-50, 50),
+                passing_users=int(cumulative_count * pass_rate))
+
+        progress.update(1)
+        progress.close()
+        logger.info("Done!")
+
+    def generate_weekly_data(self, course_id, start_date, end_date, database, delete_data=True):
+        activity_types = ['PLAYED_VIDEO', 'ATTEMPTED_PROBLEM', 'POSTED_FORUM']
+
+        # Ensure we start on a Sunday 00:00
+        days_ahead = -start_date.weekday()
+        start = start_date + datetime.timedelta(days_ahead)
+
+        if delete_data:
+            models.CourseActivityWeekly.objects.using(database).all().delete()
+            logger.info("Deleted all weekly course activity.")
+
+        logger.info("Generating new weekly course activity data...")
+
+        progress = tqdm(total=math.ceil((end_date - start).days / 7.0))
+        while start < end_date:
+            active_students = random.randint(100, 4000)
+            # End date should occur on Saturday at 23:59:59
+            end = start + datetime.timedelta(weeks=1)
```

Comment:
the new snowflake data generates weeks as monday to sunday
it also generates start and end at midnight, not 23:59:59
ie. 2019-07-07 00:00:00.000 to 2019-07-14 00:00:00.0000
so this needs a what is a week arg
or maybe make it run wednesday to wednesday to be Deliberately Fake
and in either case let's not use 59:59

Reply:
Taking the wednesday-wednesday approach